File: Habitica/src/main/java/com/habitrpg/android/habitica/models/inventory/QuestBoss.java
package com.habitrpg.android.habitica.models.inventory;


import io.realm.RealmObject;
import io.realm.annotations.Ignore;
import io.realm.annotations.PrimaryKey;

public class QuestBoss extends RealmObject {

    @PrimaryKey
    private String key;
    public String name;
    public int hp;
    public float str;

    public QuestBossRage rage;

    public boolean hasRage() {
        return rage != null && rage.value != 0;
    }

    public void setKey(String key) {
        this.key = key;
        if (rage != null) {
            rage.key = key;
        }
    }
}

File: Habitica/src/main/java/com/habitrpg/android/habitica/models/inventory/QuestBoss.kt
package com.habitrpg.android.habitica.models.inventory

import com.habitrpg.android.habitica.models.BaseObject
import io.realm.RealmObject
import io.realm.annotations.PrimaryKey

open class QuestBoss : RealmObject(), BaseObject {
    @PrimaryKey
    var key: String? = null
        set(value) {
            field = value
            rage?.key = key
        }
    var name: String? = null
    var hp: Int = 0
    var str: Float = 0.toFloat()

    var rage: QuestBossRage? = null

    val hasRage: Boolean
        get() {
            return (rage?.value ?: 0.0) > 0.0
        }
}
